src/domain/community/community/community.service.ts

```diff
@@ -422,54 +458,70 @@ export class CommunityService {
     community: ICommunity,
```

Comment:
This is one method that is too long and can easily be split into multiple units. As a rule of thumb, try having the body of your methods <20 LOC (and I know the codebase is not like that everywhere). Here you can split e.g.

```typescript 
if (contributorType == CommunityContributorType.USER) validateUserCommunityPolicy(...)
if (contributorType == CommunityContributorType.ORGANIZATION) validateOrganizationCommunityPolicy(...)
...

It can be further split on ADD REMOVE, but that is not necessary in my opinion. As a rule of thumb, if a method can't be caputred in a single screen, it's not a good sign, and I have a 25" high resolution screen and couldn't see the whole method.